### PR TITLE
Add Windows build

### PR DESCRIPTION
**Pull request by isaacbrodsky**
In addition to the usual JDK + Maven, this depends on Git, CMake, and Visual Studio. CMake is used to invoke the build system rather than calling Make directly.

Originally I tried using a shell script very similar to the *nix build script, but I had difficulty testing that in Appveyor. Instead, Powershell is used on Windows. A future improvement would be to make it easier for the user to select which MSVC version should be used in the build script.

### CODE REVIEW

File: src/main/c/h3-java/CMakeLists.txt [hunk 1]
```diff
@@ -49,7 +49,13 @@ set(JNI_SOURCE_FILES
     ${PROJECT_SOURCE_DIR}/src/com_uber_h3core_NativeMethods.h)
 
 add_library(h3-java SHARED ${JNI_SOURCE_FILES})
-target_link_libraries(h3-java ${H3_BUILD_ROOT}/lib/libh3${CMAKE_STATIC_LIBRARY_SUFFIX})
+
+# This is cached so the Windows build can indicate a different location
+# for the native core library.
+set(H3_CORE_LIBRARY_PATH "/lib/libh3" CACHE STRING "Path to the built core lubrary, without suffix")
```
Comment (nrabinowitz) [inline src/main/c/h3-java/CMakeLists.txt]: Typo: `library`

File: src/main/c/h3-java/build-h3-windows.sh [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        "x86") GENERATOR="Visual Studio 15 2017" ;;
+    esac
+
+    mkdir -p "h3-java-build$CONFIGURATION"
+    pushd "h3-java-build$CONFIGURATION"
+
+    mkdir -p build
+    pushd build
+
+    cmake -G "$GENERATOR" \
```
Comment (nrabinowitz) [inline src/main/c/h3-java/build-h3-windows.sh]: Would it be worth downloading CMake if not found? We do this in h3-py: https://github.com/uber/h3-py/blob/master/.install.sh#L22

File: src/main/c/h3-java/build-h3-windows.sh [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        -DBUILD_SHARED_LIBS=OFF \
+        -DCMAKE_BUILD_TYPE=Release \
+        ../../h3
+    cmake --build . --target h3 --config Release
+    cmake --build . --target binding-functions --config Release
+    H3_BUILD_ROOT="$(pwd)"
+
+    popd #build
```
Comment (nrabinowitz) [inline src/main/c/h3-java/build-h3-windows.sh]: Do the comments indicating the directory you're popping or the one you're now in?

File: src/main/c/h3-java/build-h3.sh [hunk 4]
```diff
@@ -67,21 +67,25 @@ pushd build
 cmake -DBUILD_SHARED_LIBS=OFF \
     -DCMAKE_BUILD_TYPE=Release \
     ../../h3
-make h3 binding-functions
+cmake --build . --target h3 --config Release
```
Comment (nrabinowitz) [inline src/main/c/h3-java/build-h3.sh]: Why wasn't this needed previously?

File: src/main/c/h3-java/build-h3-windows.sh (hunk 2 above)
Comment (isaachier) [reply src/main/c/h3-java/build-h3-windows.sh]: Interesting suggestion. I guess it depends on which decision is more user-friendly. Installing CMake as part of the installation might be more troublesome than requiring the user install it once globally.

File: src/main/c/h3-java/build-h3-windows.sh [hunk 5]
```diff
@@ -69,6 +69,7 @@ for CONFIGURATION in "x64" "x86"; do
 
     cmake -G "$GENERATOR" \
         -DBUILD_SHARED_LIBS=OFF \
+        -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
```
Comment (isaachier) [inline src/main/c/h3-java/build-h3-windows.sh]: Cool. This is definitely a good idea for the Java, Python, etc. libraries. Strictly speaking, C does not need PIC, so I'm glad we are not changing the C core to use PIC.

**Pull request merged**